pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -342,7 +342,7 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
 				uncertainNode = derr.CurrentNode
 			}
 			addErr := actualStateOfWorld.MarkVolumeAsUncertain(
-				v1.UniqueVolumeName(""),
+				volumeToAttach.VolumeName,
```

Comment:
I dont understand why we are not passing the VolumeName instead create an empty name here. This prevents us from marking an unattachable volume as `Uncertain`. I did some manual tests locally and find change it this way causes no problem. Please let me know if there is some special reason behind the empty name.

Reply:
@jingxu97 can you recall why this was empty?

Reply:
I dig some old code, and think it might be historical reason. Before, MarkVolumeAsAttached has no uniqueVolumeName parameter. It only passes volume spec and use spec to find out the unique name. Later we added this parameter to fix some issue. It seems like we just continue to use volume spec to find out the volume name instead of passing it directly. 
So far I don't see an issue directly setting it to volume name.

Reply:
If that's the case, It should be safe to use the volumeName directly. Let's see the results of all e2e tests and unit tests to see if that changes the statement.